src/dock/Dock.cpp

```diff
@@ -0,0 +1,81 @@
+// SPDX-FileCopyrightText: 2022 Nheko Contributors
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "dock/Dock.h"
+#include <qobject.h>
+#if defined(NHEKO_DBUS_SYS)
+#include <qdbusconnectioninterface.h>
+Dock::Dock(QObject *parent)
+  : QObject(parent)
+  , unityServiceWatcher(new QDBusServiceWatcher(this))
+{
+    unityServiceWatcher->setConnection(QDBusConnection::sessionBus());
+    unityServiceWatcher->setWatchMode(QDBusServiceWatcher::WatchForUnregistration |
+                                      QDBusServiceWatcher::WatchForRegistration);
+    unityServiceWatcher->addWatchedService(QStringLiteral("com.canonical.Unity"));
+    connect(unityServiceWatcher,
+            &QDBusServiceWatcher::serviceRegistered,
+            this,
+            [this](const QString &service) {
+                Q_UNUSED(service);
+                unityServiceAvailable = true;
+            });
+    connect(unityServiceWatcher,
+            &QDBusServiceWatcher::serviceUnregistered,
+            this,
+            [this](const QString &service) {
+                Q_UNUSED(service);
+                unityServiceAvailable = false;
+            });
+    QDBusPendingCall listNamesCall =
+      QDBusConnection::sessionBus().interface()->asyncCall(QStringLiteral("ListNames"));
+    QDBusPendingCallWatcher *callWatcher = new QDBusPendingCallWatcher(listNamesCall, this);
+    connect(callWatcher,
+            &QDBusPendingCallWatcher::finished,
+            this,
+            [this](QDBusPendingCallWatcher *watcher) {
+                QDBusPendingReply<QStringList> reply = *watcher;
+                watcher->deleteLater();
+
+                if (reply.isError()) {
+                    return;
+                }
+
+                const QStringList &services = reply.value();
+
+                unityServiceAvailable = services.contains(QLatin1String("com.canonical.Unity"));
+            });
+}
+
+void
+Dock::setUnreadCount(const int count)
+{
+    unitySetNotificationCount(count);
+}
+void
+Dock::unitySetNotificationCount(const int count)
+{
+    if (unityServiceAvailable) {
+        const QString launcherId = QLatin1String("nheko.desktop");
```

Comment:
This should probably be using https://doc.qt.io/qt-5/qguiapplication.html#desktopFileName-prop

Note that the function returns the name without the .desktop appended, afaik.

(The flatpak uses io.github.NhekoReborn.nheko.desktop instead, so you should use the function call instead of hardcoding it.)